### PR TITLE
[PT FE] Implement override_all_inputs, override_all_outputs

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Implement override_all_inputs, override_all_outputs*

### Tickets:
 - *119600*

### CODE REVIEW

File: src/frontends/pytorch/src/input_model.cpp [hunk 1]
```diff
@@ -81,68 +75,117 @@ void InputModel::set_partial_shape(const Place::Ptr& place, const ov::PartialSha
...
+                                    m_inputs.size() - 1,
+                                    " recieved ",
+                                    inputs.size());
+            auto self_place = m_inputs[0];
+            // Verify that no same place already in vector
+            auto no_self = std::all_of(inputs.cbegin(), inputs.cend(), [&](const Place::Ptr& p) {
+                return !p->is_equal(self_place);
+            });
+            FRONT_END_GENERAL_CHECK(no_self,
+                                    "Unexpected input place of 'self' input was provided to override_all_inputs.");
```
Comment (PiotrKrzem) [inline src/frontends/pytorch/src/input_model.cpp:166]: Should this be "Unexpected input place of 'self'. Input was provided to override_all_inputs."?

File: src/frontends/pytorch/src/place.hpp [hunk 2]
```diff
@@ -11,7 +11,12 @@ namespace ov {
 namespace frontend {
 namespace pytorch {
 
+class InputModel;
+
+// Place represent a tensor in the model
```
Comment (PiotrKrzem) [inline src/frontends/pytorch/src/place.hpp:16]:
```suggestion
// Place represents a tensor in the model
```

**Review by PiotrKrzem — APPROVED** [review]

File: src/frontends/pytorch/src/input_model.cpp (hunk 1 above)
Comment (mvafin) [reply src/frontends/pytorch/src/input_model.cpp:166]: No, but agree, this is complicated sentence. The subject here is "input place" that represent "self input", I will simplify it.

Comment (mvafin) [inline src/frontends/pytorch/src/input_model.cpp:166]:
```suggestion
                                    "Unexpected input of 'self' was provided to override_all_inputs.");
```

File: src/frontends/pytorch/src/input_model.cpp [hunk 3]
```diff
@@ -81,68 +75,117 @@ void InputModel::set_partial_shape(const Place::Ptr& place, const ov::PartialSha
...
+                                    m_inputs.size() - 1,
+                                    " recieved ",
+                                    inputs.size());
+            auto self_place = m_inputs[0];
+            // Verify that no same place already in vector
+            auto no_self = std::all_of(inputs.cbegin(), inputs.cend(), [&](const Place::Ptr& p) {
+                return !p->is_equal(self_place);
+            });
+            FRONT_END_GENERAL_CHECK(no_self,
+                                    "Unexpected input of 'self' was provided to override_all_inputs.");
```
Comment (mvafin) [inline src/frontends/pytorch/src/input_model.cpp:166]:
```suggestion
            FRONT_END_GENERAL_CHECK(no_self, "Unexpected input of 'self' was provided to override_all_inputs.");
```

**Pull request merged**